### PR TITLE
Add 4 Bessel Special Operators

### PR DESCRIPTION
**Pull request by Kanya-Mo**
- [x] special_bessel_j0.out
- [x] special_bessel_j0
- [x] special_bessel_j1.out
- [x] special_bessel_j1
- [x] special_bessel_y0.out
- [x] special_bessel_y0
- [x] special_bessel_y1.out
- [x] special_bessel_y1

### CODE REVIEW

File: test/xpu/xpu_test_utils.py [hunk 1]
```diff
@@ -252,6 +252,10 @@
     "square",
     "heaviside",
     "argsort",
+    "special_bessel_j0",
+    "special_bessel_j1",
```
Comment (majing921201) [inline test/xpu/xpu_test_utils.py:256]: Those operators are not in common_methods_invocations, don't need to add here. please remove.

**Comment by xytintel** [comment]
`aten/src/ATen/native/Math.h` can be reused

File: src/ATen/native/xpu/sycl/MathExtensions.h [hunk 2]
```diff
@@ -361,4 +361,240 @@ static inline C10_HOST_DEVICE scalar_t calc_i1e(scalar_t _x) {
   return (_x < scalar_t{0.0}) ? -out : out;
 }
 
+template <typename scalar_t>
+static inline C10_HOST_DEVICE scalar_t bessel_j1_forward(scalar_t x) {
```
[the remark this answers is not in the document]
Comment (majing921201) [reply src/ATen/native/xpu/sycl/MathExtensions.h:365]: > Why no `bessel_j0_forward` and `bessel_y0_forward` here?

They are in `torch/testing/_internal/opinfo/definitions/special.py`, Our datatype extended test only convers ops in `common_methods_invocations.py`
I think PyTorch test will cover

**Comment by Kanya-Mo** [comment]
> `aten/src/ATen/native/Math.h` can be reused for `bessel_j1_forward` and `bessel_y1_forward`.

Hi @xytintel, the implementation of `bessel_j1_forward` and `bessel_y1_forward` in aten/src/ATen/native/Math.h include recursive call for some input which is not supported by sycl (compilation will fail with a message 'SYCL kernel cannot call a recursive function"), so I also upstreamed their ipex implementation. On the other hand, torch's `bessel_j0_forward` and `bessel_y0_forward` are fine and that's why I didn't include them in MathExtension.h.

File: test/xpu/xpu_test_utils.py (hunk 1 above)
Comment (Kanya-Mo) [reply test/xpu/xpu_test_utils.py:256]: Thanks for pointing out. Will remove.

**Review by xytintel — APPROVED** [review]

**Pull request merged**